From 35164419521589472692efbc00037299a7ed7ad0 Mon Sep 17 00:00:00 2001
From: Chris Talbot <chris@talbothome.com>
Date: Wed, 14 Feb 2024 14:15:39 -0700
Subject: [PATCH] test: disable sip test

---
 plugins/provider/tests/meson.build | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/plugins/provider/tests/meson.build b/plugins/provider/tests/meson.build
index 49f5cd1..25ef99e 100644
--- a/plugins/provider/tests/meson.build
+++ b/plugins/provider/tests/meson.build
@@ -95,7 +95,7 @@ t = executable('media', test_sources,
                  ]
                )
 test('media', t, env: test_env)
-
+'''
 test_sources = [ 'test-sip.c' ]
 test_sources += sip_sources
 t = executable('sip', test_sources,
@@ -110,7 +110,7 @@ t = executable('sip', test_sources,
                  ]
                )
 test('sip', t, env: test_env)
-
+'''
 test_sources = [ 'test-srtp.c' ]
 test_sources += sip_sources
 t = executable('srtp', test_sources,
-- 
2.43.0

